src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -78,17 +65,28 @@ public T remove(int index) {
 
     @Override
     public boolean remove(T object) {
-        int index = -1;
-        for (int i = 0; i < size; i++) {
-            T element = get(i);
-            if (element == object || (element != null && element.equals(object))) {
-                index = i;
-                break;
+        if (size != 0) {
+            int index = -1;
+            Node<T> currNode = head;
+            for (int i = 0; i < size; i++) {
+                if (currNode.item == object
+                        || (currNode.item != null && currNode.item.equals(object))) {
+                    index = i;
+                    break;
+                }
+                currNode = currNode.next;
+            }
+            if (index != -1) {
+                if (index == 0) {
+                    removeFirst();
+                } else if (index == size - 1) {
+                    removeLast();
+                } else {
+                    unlink(currNode);
+                    size--;
+                }
+                return true;
```

Comment:
Why can't you use existing logic from the remove method?

Reply:
Because in remove method I know the index, so I can apply a binary search algorithm. Here I have no clue where my node is. And in the worst case scenario, I would need to visit all nodes anyway. 
Or maybe I did not get what you meant by "logic"